#! /bin/sh /usr/share/dpatch/dpatch-run
## 04rpath_fix.dpatch by  <bernat@neo.luffy.cx>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix rpath issues in Makefiles

@DPATCH@

--- xrdp-0.4.0.orig/sesman/tools/Makefile
+++ xrdp-0.4.0/sesman/tools/Makefile
@@ -14,7 +14,7 @@
 DEFINES = -DLIBSCP_CLIENT
 
 CFLAGS = -Wall -O2 -I../../common -I../ -I/usr/include/nptl -I../libscp $(DEFINES)
-LDFLAGS = -L/usr/gnu/lib -L/usr/lib/nptl -L../libscp -Wl,-rpath,. -lpthread -ldl -lscp
+LDFLAGS = -L/usr/gnu/lib -L/usr/lib/nptl -L../libscp -Wl,-rpath,$(DESTDIR) -lpthread -ldl -lscp
 C_OS_FLAGS = $(CFLAGS) -c
 CC = gcc
 
--- xrdp-0.4.0.orig/sesman/Makefile
+++ xrdp-0.4.0/sesman/Makefile
@@ -16,8 +16,9 @@
 	  -DSESMAN_PID_FILE=\"$(PIDDIR)/sesman.pid\" \
 	  -DSESMAN_SESSVC_FILE=\"sessvc\"
 
+
 CFLAGS = -Wall -O2 -I../common -I/usr/include/nptl -I./libscp $(DEFINES)
-LDFLAGS = -L/usr/gnu/lib -L/usr/lib/nptl -L./libscp -Wl,-rpath,. -lpthread -ldl -lscp
+LDFLAGS = -L/usr/gnu/lib -L/usr/lib/nptl -L./libscp -Wl,-rpath,$(DESTDIR) -lpthread -ldl -lscp
 C_OS_FLAGS = $(CFLAGS) -c
 CC = gcc
 
--- xrdp-0.4.0.orig/xrdp/Makefile
+++ xrdp-0.4.0/xrdp/Makefile
@@ -11,7 +11,7 @@
 CFLAGS = -Wall -O2 -I../common -I../libxrdp $(DEFINES)
 #CFLAGS += -DXRDP_DEBUG
 C_OS_FLAGS = $(CFLAGS) -c
-LDFLAGS = -L/usr/gnu/lib -L../libxrdp -Wl,-rpath,.
+LDFLAGS = -L/usr/gnu/lib -L../libxrdp -Wl,-rpath,$(DESTDIR)
 LIBS = -ldl -lpthread -lxrdp
 CC = gcc
 
